flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosApplicationMasterRunner.java

```diff
@@ -0,0 +1,618 @@
+				throw new RuntimeException("Invalid value for " + MesosConfigKeys.ENV_TM_COUNT + " : "
+					+ e.getMessage());
+			}
+			try {
+				slotsPerTaskManager = Integer.parseInt(ENV.get(MesosConfigKeys.ENV_SLOTS));
+			} catch (NumberFormatException e) {
+				throw new RuntimeException("Invalid value for " + MesosConfigKeys.ENV_SLOTS + " : "
+					+ e.getMessage());
+			}
+
+			final ContaineredTaskManagerParameters containeredParameters =
+				ContaineredTaskManagerParameters.create(config, taskManagerContainerMemory, slotsPerTaskManager);
+
+			final MesosTaskManagerParameters taskManagerParameters =
+				MesosTaskManagerParameters.create(config, containeredParameters);
+
+			LOG.info("TaskManagers will be created with {} task slots",
+				taskManagerParameters.containeredParameters().numSlots());
+			LOG.info("TaskManagers will be started with container size {} MB, JVM heap size {} MB, " +
+					"JVM direct memory limit {} MB, {} cpus",
+				taskManagerParameters.containeredParameters().taskManagerTotalMemoryMB(),
+				taskManagerParameters.containeredParameters().taskManagerHeapSizeMB(),
+				taskManagerParameters.containeredParameters().taskManagerDirectMemoryLimitMB(),
+				taskManagerParameters.cpus());
+
+			// JM endpoint, which should be explicitly configured by the dispatcher (based on acquired net resources)
+			final int listeningPort = config.getInteger(ConfigConstants.JOB_MANAGER_IPC_PORT_KEY,
+				ConfigConstants.DEFAULT_JOB_MANAGER_IPC_PORT);
+			require(listeningPort >= 0 && listeningPort <= 65536, "Config parameter \"" +
+				ConfigConstants.JOB_MANAGER_IPC_PORT_KEY + "\" is invalid, it must be between 0 and 65536");
+
+			// ----------------- (2) start the actor system -------------------
+
+			// try to start the actor system, JobManager and JobManager actor system
+			// using the configured address and ports
+			actorSystem = BootstrapTools.startActorSystem(config, appMasterHostname, listeningPort, LOG);
+
+			final String akkaHostname = AkkaUtils.getAddress(actorSystem).host().get();
+			final int akkaPort = (Integer) AkkaUtils.getAddress(actorSystem).port().get();
+
+			LOG.info("Actor system bound to hostname {}.", akkaHostname);
+
+			// try to start the artifact server
+			LOG.debug("Starting Artifact Server");
+			final int artifactServerPort = config.getInteger(ConfigConstants.MESOS_ARTIFACT_SERVER_PORT_KEY,
+				ConfigConstants.DEFAULT_MESOS_ARTIFACT_SERVER_PORT);
+			artifactServer = new MesosArtifactServer(sessionID, akkaHostname, artifactServerPort);
+
+			// ----------------- (3) Generate the configuration for the TaskManagers -------------------
+
+			final Configuration taskManagerConfig = BootstrapTools.generateTaskManagerConfiguration(
+				config, akkaHostname, akkaPort, slotsPerTaskManager, TASKMANAGER_REGISTRATION_TIMEOUT);
+			LOG.debug("TaskManager configuration: {}", taskManagerConfig);
+
+			final Protos.TaskInfo.Builder taskManagerContext = createTaskManagerContext(
+				config, mesosConfig, ENV,
```

Comment:
True, that is the old discussion about coding style and we won't solve this here (and most likely this won't be solved at all). But I think we can try to be consistent within one PR, for example. The thing with a variable number of arguments per line is that it might be really readable at the moment. But once we add a new parameter, you basically have to format the whole thing again, instead of adding a single line. So I think it's not optimal for reasons of maintainability. But I'm not adamant here because the Flink community does not have a standard.